src/Parser/Warlock/Destruction/Modules/Talents/Eradication.js

```diff
@@ -27,26 +28,39 @@ class Eradication extends Analyzer {
     this.active = this.combatants.selected.hasTalent(SPELLS.ERADICATION_TALENT.id) || this.combatants.selected.hasFinger(ITEMS.SOUL_OF_THE_NETHERLORD.id);
   }
 
+  suggestions(when) {
+    const uptime = this.enemies.getBuffUptime(SPELLS.ERADICATION_DEBUFF.id) / this.owner.fightDuration;
+    when(uptime).isLessThan(0.7)
+      .addSuggestion((suggest, actual, recommended) => {
+        return suggest(<span>Your <SpellLink id={SPELLS.ERADICATION_DEBUFF.id} /> uptime can be improved. Try to spread your chaos bolts out for higher uptime.</span>)
+          .icon(SPELLS.ERADICATION_DEBUFF.icon)
+          .actual(`${formatPercentage(actual)}% Eradication uptime`)
+          .recommended(`>${formatPercentage(recommended)}% is recommended`)
+          .regular(recommended - 0.05).major(recommended - 0.10);
+      });
+  }
+
   on_byPlayer_damage(event) {
     const enemy = this.enemies.getEntity(event);
-    if (!enemy || enemy.hasBuff(SPELLS.ERADICATION_DEBUFF.id, event.timestamp)) {
+    if (!enemy || !enemy.hasBuff(SPELLS.ERADICATION_DEBUFF.id, event.timestamp)) {
       return;
     }
     this.bonusDmg += getDamageBonus(event, ERADICATION_DAMAGE_BONUS);
   }
 
-  statistic() {
+  statistic() {    
+    const uptime = this.enemies.getBuffUptime(SPELLS.ERADICATION_DEBUFF.id) / this.owner.fightDuration;
     return (
       <StatisticBox
         icon={<SpellIcon id={SPELLS.ERADICATION_TALENT.id} />}
-        value={`${formatNumber(this.bonusDmg / this.owner.fightDuration * 1000)} DPS`}
-        label="Damage contributed"
+        value={`${formatPercentage(uptime)} %`}
+        label="Eradication Uptime"
```

Comment:
For talents, we typically like to express the effectiveness of the talent in terms of percentage damage gained, so that it can be used as a point of reference vs. other talents in the same tier. That's why this showed damage instead of uptime. It could be worthwhile to show both, or one in the stats box and the other in the tooltip.

Reply:
that’s what I did, the damage is still in the tooltip, but it’s more helpful to see uptime in the box.